### ISSUE TITLE
Investiage RPC connection lifetime issues

### ISSUE THREAD

**Issue by MarcoFalke** [bitcoin/bitcoin#18623]
At least our tests are plagued by RPC connection lifetime issues. The connection (in python, `test/functional/test_framework/authproxy.py`) often throws `BrokenPipeError` ( `ConnectionResetError` on freebsd, `OSError` on Windows)

On Windows, running feature_fee_estimation is just not possible at all. See https://github.com/bitcoin/bitcoin/blob/d9fd7b5a675d94b99a5ee5b636bbc21088ed02fb/.appveyor.yml#L88

#### Useful skills:

* Strong background in internet protocols
* Strong background in libevent
* Basic knowledge of the Bitcoin Core HTTP RPC server and functional tests' RPC dispatcher

#### Want to work on this issue?

The purpose of the `good first issue` label is to highlight which issues are suitable for a new contributor without a deep understanding of the codebase.

You do not need to request permission to start working on this. You are encouraged to comment on the issue if you are planning to work on it. This will help other contributors monitor which issues are actively being addressed and is also an effective way to request assistance if and when you need it.

For guidance on contributing, please read [CONTRIBUTING.md](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md) before opening your pull request.

**Comment by promag** [comment]
Example can be found here https://ci.appveyor.com/project/DrahtBot/bitcoin/builds/32122845#L4916.

**Comment by MarcoFalke** [comment]
The issues have been worked around in #18596, so they should only happen invisibly. Still, reason enough to fix them.

**Comment by MarcoFalke** [comment]
The functional tests have been disabled on windows for now. See fa50d11dd117eb6f91cbeac7476797f3885a6169

**Comment by kilotes** [comment]
.

**Comment by kanzure** [comment]
Retrying RPC requests can be dangerous outside of the tests... Someone recently suggested a nonce: https://github.com/bitcoin/bitcoin/issues/9197

**Comment by MarcoFalke** [comment]
Yes, and my worry is that some users who run into this in production will copy our test authproxy and think that it is the "official" one.

**Comment by willcl-ark** [comment]
I was just looking at this issue, and it seems that the historical context was actually to _remove_ HTTP `keep-alive` header (behind a flag) in 16a5c18cea7330bd68dc9d2f768eb518af88795b due to long-lived connection issues, and then finally erase it completely in the RPC overhaul of 40b556d3742a1f65d67e2d4c760d0b13fe8be5b7.

I am therefore wondering, would the aim of this issue be to re-introduce `keep-alive` again (behind a flag, defaulted to `off`) which could be used by the test framework, or to try and find an alternative approach?
